test/conn_test.go

```diff
@@ -817,6 +824,15 @@ func TestCallbacksOrder(t *testing.T) {
 	if rtime.Before(dtime1) || dtime2.Before(rtime) || atime2.Before(atime1) || ctime.Before(atime2) {
 		t.Fatalf("Wrong callback order:\n%v\n%v\n%v\n%v\n%v\n%v", dtime1, rtime, atime1, atime2, dtime2, ctime)
 	}
+
+	// Close the other connection
+	ncp.Close()
+
+	// Give a chance for the go routine to stop
+	time.Sleep(150 * time.Millisecond)
```

Comment:
Be good if this was deterministic instead of time based.

Reply:
I could add specific wait group in the struct and use that for the test, but I feel it is a bit too much. What I could do is repeat the test with a little delay in case of failure and fail the test after few attempts. I will do that.